dot/network/notifications.go

```diff
@@ -247,6 +242,15 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 			logger.Trace("sending handshake", "protocol", info.protocolID, "peer", peer, "message", hs)
 			err = s.host.send(peer, info.protocolID, hs)
 		} else {
+			if s.host.messageCache != nil {
+				var added bool
+				added, err = s.host.messageCache.Put(peer, msg.String())
+				if err != nil || !added {
+					logger.Error("failed to add message to cache", "peer", peer, "error", err)
```

Comment:
will `Put` error if the key already exists?

Reply:
No. It will return `false`.

Reply:
in this case, it will log the error if the message is already in the cache, maybe remove the `!added` case for now

Reply:
Done